src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -464,6 +659,38 @@ def create_or_upgrade_app(
                         )
                         print_messages(console, upgrade_cursor)
 
+                        events_definitions = (
+                            get_snowflake_facade().get_event_definitions(
+                                app_name=self.name
+                            )
+                        )
+
+                        app_properties = get_snowflake_facade().desc_application(
+                            self.name
+                        )
+                        new_authorize_event_sharing_value = (
+                            event_sharing.should_authorize_event_sharing(
+                                app_properties,
+                                events_definitions,
+                            )
+                        )
+                        if new_authorize_event_sharing_value is not None:
+                            log.info(
+                                "Setting telemetry sharing authorization to %s",
+                                new_authorize_event_sharing_value,
+                            )
+                            sql_executor.execute_query(
+                                f"alter application {self.name} set AUTHORIZE_TELEMETRY_EVENT_SHARING = {str(new_authorize_event_sharing_value).upper()}"
+                            )
+                        events_to_share = event_sharing.events_to_share(
+                            events_definitions
+                        )
```

Comment:
to be honest, this event sharing class is a bit odd to me, it only encapsulates some of the state but not all? I haven't fully wrapped my head around it.